### PR TITLE
Change onClick of dropdown to onDropdownVisiblityChange

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

**Review by tnagorra — APPROVED** [review]

File: components/Input/ColorInput/index.js [hunk 1]
```diff
@@ -94,26 +110,37 @@ const defaultProps = {
...
+        showSwatches,
+    } = props;
 
-    static defaultProps = defaultProps;
+    const [showColorPicker, setShowColorPicker] = useState(false);
+    const [boundingClientRect, setBoundingClientRect] = useState({});
+    const containerRef = useRef(undefined);
+    const [selectedScheme, setSelectedScheme] = useState('sequential');
+    // NOTE: In color brewer each scheme group has at least one color swatch with 8 items
+    const [selectedNumberOfColors, setSelectedNumberOfColors] = useState('8');
```
Comment (tnagorra) [inline components/Input/ColorInput/index.js]: Value for selectedNumberOfColors is string but the value from options is a number

File: components/Input/ColorInput/index.js [hunk 2]
```diff
@@ -133,120 +160,143 @@ class ColorInput extends React.PureComponent {
...
+
+    const swatchesColors = useMemo(() => {
+        const schemes = colorBrewer.schemeGroups[selectedScheme];
+        const colorsForSwatches = [];
+        schemes.forEach((scheme) => {
+            const colorsForScheme = colorBrewer[scheme];
+            if (colorsForScheme[selectedNumberOfColors]) {
+                colorsForSwatches.push(colorsForScheme[selectedNumberOfColors]);
+            }
+        });
```
Comment (tnagorra) [inline components/Input/ColorInput/index.js]: const colorsForSwatches = schemas.map(scheme => colorsForScheme[scheme][selectedNumberOfColors]).filter(isDefined);

File: components/Input/ColorInput/index.js [hunk 3]
```diff
@@ -1,25 +1,95 @@
...
+    const {
+        onChange,
+        value,
+        swatches,
+    } = props;
+
+    return (
+        <div className={styles.swatchesPicker}>
+            {swatches.map(swatch => (
+                <div className={styles.swatch}>
```
Comment (tnagorra) [inline components/Input/ColorInput/index.js]: Need to add key prop

File: components/Input/ColorInput/index.js [hunk 4]
```diff
@@ -1,25 +1,95 @@
...
+                    {swatch.map(color => (
+                        <Button
```
Comment (tnagorra) [inline components/Input/ColorInput/index.js]: Need to add key prop

File: components/Input/ColorInput/index.js [hunk 5]
```diff
@@ -1,25 +1,162 @@
...
+        color: data,
+        value,
+        onColorChange,
+    }), [onColorChange, value]);
+
+    return (
+        <ListView
+            className={styles.swatch}
+            data={colors}
+            keySelector={identitySelector}
```
Comment (tnagorra) [inline components/Input/ColorInput/index.js:92]: This doesn't work.
key should be string or number

File: components/Input/ColorInput/index.js [hunk 6]
```diff
@@ -1,25 +1,162 @@
...
+    value: PropTypes.string,
+    colors: PropTypes.arrayOf(PropTypes.string).isRequired,
+    onColorChange: PropTypes.func.isRequired,
+};
+
+Swatch.defaultProps = {
+    value: undefined,
+};
+
+const swatchKeySelector = d => d.join(',');
```
Comment (tnagorra) [inline components/Input/ColorInput/index.js:109]: Is this a hack?

Comment (AdityaKhatri) [reply components/Input/ColorInput/index.js:109]: yes and no

**Pull request merged**